modules/gltf/src/lib/extensions/EXT_mesh_features.ts

```diff
@@ -91,3 +100,118 @@ function processMeshPrimitiveFeatures(
     featureId.data = featureIdData;
   }
 }
+
+/*
+  Encoding data
+*/
+
+function encodeExtMeshFeatures(scenegraph: GLTFScenegraph, options: GLTFWriterOptions) {
+  const meshes = scenegraph.gltf.json.meshes;
+  if (!meshes) {
+    return;
+  }
+
+  // Iterate through all meshes/primitives.
+  for (const mesh of meshes) {
+    for (const primitive of mesh.primitives) {
+      encodeExtMeshFeaturesForPrimitive(scenegraph, primitive);
+    }
+  }
+}
+
+/**
+ * Creates ExtMeshFeatures, creates a featureId containing feature ids provided.
+ * @param scenegraph - Instance of the class for structured access to GLTF data.
+ * @param primitive - target primitive instance that will contain the extension
+ * @param featureIdArray - Array of feature id
+ * @param propertyTableIndex - index of the property table created by the ExtStructuralMetadata (optional).
+ */
+export function createExtMeshFeatures(
+  scenegraph: GLTFScenegraph,
+  primitive: GLTFMeshPrimitive,
+  featureIdArray: NumericArray,
+  propertyTableIndex?: number
+) {
+  if (!primitive.extensions) {
+    primitive.extensions = {};
+  }
+  let extension = primitive.extensions[EXT_MESH_FEATURES_NAME] as GLTF_EXT_mesh_features;
+  if (!extension) {
+    extension = {featureIds: []};
+    primitive.extensions[EXT_MESH_FEATURES_NAME] = extension;
+  }
+
+  const {featureIds} = extension;
+  const featureId: GLTF_EXT_mesh_features_featureId = {
+    featureCount: featureIdArray.length,
+    propertyTable: propertyTableIndex,
+    data: featureIdArray
+  };
+  featureIds.push(featureId);
+
+  scenegraph.addObjectExtension(primitive, EXT_MESH_FEATURES_NAME, extension);
+}
+
+/**
+ * Encodes a feature ID set to extension.
+ * @param scenegraph - Instance of the class for structured access to GLTF data.
+ * @param primitive - Primitive that the data encoded belongs to.
+ * @see https://github.com/CesiumGS/glTF/tree/3d-tiles-next/extensions/2.0/Vendor/EXT_mesh_features
+ */
+function encodeExtMeshFeaturesForPrimitive(
+  scenegraph: GLTFScenegraph,
+  primitive: GLTFMeshPrimitive
+) {
+  const extension = primitive.extensions?.[EXT_MESH_FEATURES_NAME] as GLTF_EXT_mesh_features;
+  if (!extension) {
+    return;
+  }
+  const featureIds: GLTF_EXT_mesh_features_featureId[] = extension.featureIds;
+  featureIds.forEach((featureId, ind) => {
```

Comment:
Type out variable names
```suggestion
  featureIds.forEach((featureId, index) => {
```

Reply:
Done.